basiclti/basiclti-common/src/java/org/sakaiproject/basiclti/util/SakaiBLTIUtil.java

```diff
@@ -532,10 +535,39 @@ public static void addUserInfo(Properties ltiProps, Properties lti13subst, Map<S
 		}
 	}
 
+	public static String upgradeRoleString(String roleString)
+	{
+		if ( roleString == null ) return null;
+		if ( roleString.length() < 1 ) return roleString;
```

Comment:
if (StringUtils.isBlank()) return StringUtils.EMPTY; ? That way you always get at least an empty string, never null, and you don't need null checks when you're splitting the result.

Reply:
Thanks for catching this - one day I will write this code automatically.